hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/BlockRecordManagerImpl.java

```diff
@@ -0,0 +1,482 @@
+    /** The hash size in bytes, normally 48 for SHA384 */
+    private static final int HASH_SIZE = DigestType.SHA_384.digestLength();
+
+    /** Number of blocks to keep multiplied by hash size */
+    private final int numBlockHashesToKeepBytes;
+    /** The number of milliseconds of consensus time in a block period, from configuration */
+    private final long blockPeriodInMilliSeconds;
+    /** The configuration provider */
+    private final ConfigProvider configProvider;
+    /** The stream file producer we are using */
+    private final StreamFileProducerBase streamFileProducer;
+    /** The working state accessor, for accessing state adhoc */
+    private final WorkingStateAccessor workingStateAccessor;
+    /** Ture when we have completed event recovery */
+    private boolean eventRecoveryCompleted = false;
+    /** The number of the current block. "provisional" because the block is not yet complete. */
+    private long provisionalCurrentBlockNumber;
+    /** The consensus time of the first transaction in the current block. "provisional" because the block is not yet complete. */
+    private Instant provisionalCurrentBlockFirstTransactionTime = null;
+
+    /**
+     * Construct BlockRecordManager
+     *
+     * @param configProvider The configuration provider
+     * @param nodeInfo The node info
+     * @param signer The signer for signing files
+     * @param fileSystem the file system to use, needed for testing to be able to use a non-standard file
+     *                   system. If null default is used.
+     * @param workingStateAccessor The working state accessor, for accessing state adhoc
+     */
+    @Inject
+    public BlockRecordManagerImpl(
+            @NonNull final ConfigProvider configProvider,
+            @NonNull final NodeInfo nodeInfo,
+            @NonNull final Signer signer,
+            @Nullable final FileSystem fileSystem,
+            @NonNull final WorkingStateAccessor workingStateAccessor) {
+        this.configProvider = requireNonNull(configProvider);
+        this.workingStateAccessor = requireNonNull(workingStateAccessor);
+        this.streamFileProducer = new StreamFileProducerConcurrent(
+                configProvider,
+                requireNonNull(nodeInfo),
+                requireNonNull(signer),
+                fileSystem,
+                ForkJoinPool.commonPool());
+        // check if we were started in event recover mode and if event recovery needs to be completed before we write
+        // any new records to stream
+        this.eventRecoveryCompleted = !nodeInfo.wasStartedInEventStreamRecovery();
+        // get configuration needed, this is configuration that is assumed not to change while the node is running
+        BlockRecordStreamConfig recordStreamConfig =
+                configProvider.getConfiguration().getConfigData(BlockRecordStreamConfig.class);
+        this.blockPeriodInMilliSeconds = recordStreamConfig.logPeriod() * 1000L;
```

Comment:
What should happen, if this configuration is changed?

Reply:
See answer above

Reply:
I agree with you, but I am afraid we have no choice, if we want to be 100% consistent.